browser/brave_tab_helpers.cc

```diff
@@ -89,13 +89,19 @@
 #include "brave/browser/playlist/playlist_tab_helper.h"
 #endif
 
+namespace brave {
+
 #if defined(TOOLKIT_VIEWS)
-#include "brave/browser/ui/sidebar/sidebar_tab_helper.h"
+// Register per-tab(contextual) side-panel registry.
+// Defined at //brave/browser/ui/views/side_panel/brave_side_panel_utils.cc as
+// the implementation is view-layer specific.
+void RegisterContextualSidePanel(content::WebContents* web_contents);
 #endif
 
-namespace brave {
-
 void AttachTabHelpers(content::WebContents* web_contents) {
+#if defined(TOOLKIT_VIEWS)
+  RegisterContextualSidePanel(web_contents);
```

Comment:
I don't think this makes sense in `AttachTabHelpers` because it's not adding tab helpers. Where does this happen in upstream?

Reply:
Usually, upstream has tab helper for each contextual side panel. And that tab helper registers its `SidePanelEntry`.
In our case, we also could have tab helpr for this but only its duty is registring sidepanel.
So, called this at here.

Reply:
`AttachTabHelpers` is not just a generic place to do things when webcontents are created, it is specifically for TabHelpers so we should not be calling this here

Reply:
https://github.com/brave/brave-browser/issues/38886